sshd/src/test/scala/ammonite/repl/sshd/SshdReplTest.scala

```diff
@@ -0,0 +1,48 @@
+package ammonite.repl.sshd
+
+import utest._
+import utest.framework.TestSuite
+
+import scala.concurrent.{Await, Promise}
+import scala.language.postfixOps
+import scala.concurrent.duration._
+
+object SshdReplTest extends TestSuite with SshTestingTraits {
+  val remotePromise = Promise[Boolean]()
+
+  override val tests = TestSuite {
+    'canExecuteRemoteCommand{
+      withTmpDirectory { tmpDir ⇒
+        val repl = new SshdRepl(SshServerConfig(0, testUsername, testPassword, tmpDir))
+        repl.start()
+        val client = sshClient((testUsername, testPassword), "localhost", repl.port)
+        client.connect()
+        assert(client.isConnected)
+
+        val shell = new Shell(client)
+        shell.connect()
+        assert(shell.isConnected)
+
+        shell.input.println("2 + 2")
+        shell.input.println("import ammonite.repl.sshd.SshdReplTest")
+        shell.input.println("SshdReplTest.remotePromise.success(true)")
+        shell.input.println("exit")
+
+        assert(Await.result(remotePromise.future, 60 seconds))
+
+        val outputLines = shell.output.readAll.lines.toList
+        val firstBannerLine :: secondBannerLine :: Nil = outputLines.take(2)
+        val mathTestPrompt :: mathOutput :: Nil = outputLines.slice(2, 4)
+        val importAppPackage :: importResult :: Nil = outputLines.slice(4, 6)
+
+        assert(firstBannerLine.startsWith("Welcome to the Ammonite Repl"))
```

Comment:
uTest's `assert` can take multiple expressions as varargs; you can have one `assert` call and just pass each of these expressions on a new line. You also shouldn't need the `&&`s; those can just be separate expressions in the one, large `assert` call